src/libraries/System.Text.RegularExpressions/tests/RegexReductionTests.cs

```diff
@@ -371,6 +371,17 @@ private static int GetMinRequiredLength(Regex r)
         [InlineData("xyz(?:(?i:abcde)|(?i:abcdf))", "xyz(?i:abcd[ef])")]
         [InlineData("bonjour|hej|ciao|shalom|zdravo|pozdrav|hallo|hola|hello|hey|witam|tere|bonjou|salam|helo|sawubona", "(?>bonjou(?>r|)|h(?>e(?>j|(?>l(?>lo|o)|y))|allo|ola)|ciao|s(?>halom|a(?>lam|wubona))|zdravo|pozdrav|witam|tere)")]
         [InlineData("\\w\\d123|\\w\\dabc", "\\w\\d(?:123|abc)")]
+        // Loops inside alternation constructs
```

Comment:
are there enough tests ensuring this doesn't happen if there's incompatible options? eg this should not atomify:
`(a(?-i:b)(?-i:B*)|def)ghi`
and possibly the same for other options. I don't see `(?-i` in this file at all.

Reply:
With respect to options, perhaps these should have a optional global option. So
`[InlineData("abB*", "ab+", RegexOptions.IgnoreCase]` should pass?

Reply:
> With respect to options, perhaps these should have a optional global option

The options check / impact will go away https://github.com/dotnet/runtime/issues/61048; since we're ignoring RightToLeft, the only other option that's relevant here (and the reason this check exists) is IgnoreCase, and after #61048, IgnoreCase will be removed from everything except for backreferences.

Reply:
Ah, yes.  Unrelated, it does seem we're a little short on functional tests of inline enable/disable of options, I'll make a note.